qemu/tests/qmp_command.py

```diff
@@ -200,41 +160,39 @@ def check_result(qmp_o, output=None, exception_list=""):
         qmp_port = qmp_ports[0]
     else:
         test.error("Incorrect configuration, no QMP monitor found.")
-    hmp_ports = vm.get_monitors_by_type('human')
-    if hmp_ports:
-        hmp_port = hmp_ports[0]
-    else:
-        test.error("Incorrect configuration, no QMP monitor found.")
     callback = {"host_cmd": lambda cmd: process.system_output(cmd, shell=True).decode(),
                 "guest_cmd": session.cmd_output,
-                "monitor_cmd": hmp_port.send_args_cmd,
                 "qmp_cmd": qmp_port.send_args_cmd}
 
     def send_cmd(cmd):
-        """ Helper to execute command on ssh/host/monitor """
+        """ Helper to execute command on host/ssh guest/qmp monitor """
         if cmd_type in callback.keys():
             return callback[cmd_type](cmd)
         else:
             test.error("cmd_type is not supported")
 
     pre_cmd = params.get("pre_cmd")
     qmp_cmd = params.get("qmp_cmd")
-    cmd_type = params.get("event_cmd_type")
     post_cmd = params.get("post_cmd")
+    cmd_type = params.get("event_cmd_type")
     result_check = params.get("cmd_result_check")
-    cmd_return_value = params.get("cmd_return_value")
-    exception_list = params.get("exception_list", "")
+    strict_match = params.get("strict_match", "yes") == 'yes'
+    expect_o = eval(params.get("cmd_return_value", "[]"))
 
-    # Pre command
+    # Pre qmp command
```

Comment:
The pre_cmd is not always qmp command, eg. 
> pre_cmd = "rm -rf /var/tmp/memsave"

Reply:
of course, it is not.   We use `event_cmd_type` to define how to run `pre_cmd` and `post_cmd`. 
If you check `send_cmd` carefully,  you can got `callback[cmd_type](cmd)`.

Reply:
Ok, will update this comment, thanks !